applications/conceptual-model-editor/src/app/diagram/reactflow/components/drag-edge-canvas-menu.tsx

```diff
@@ -0,0 +1,52 @@
+import { XYPosition } from "reactflow";
+import { useMenuOptions } from "./menu-options";
+import { useState } from "react";
+
+export const useCanvasMenuOptions = () => {
+
+  const { 
+    MenuOptionsGeneral,
+    isMenuOptionsOpen,
+    openMenuOptions: open,
+    closeMenuOptions: close, 
+    } = useMenuOptions();
+
+    const [canvasMenuXYPosition, setCanvasMenuXYPosition] = useState<XYPosition>();
+
+    const CanvasMenuOptions = (props: {    
+            openCreateClassDialogHandler: () => void;  
+            openSelectClassDialog: () => void;
+        }) => {
+
+
+        const handlersWithTexts: Array<[(() => void) | undefined, string]> = [      
+            [props.openCreateClassDialogHandler, "➕ Create new class"],
+            [undefined, "HorizontalSeparator"],
+            [props.openSelectClassDialog, "🕮 Choose existing classes"],
+            [undefined, "HorizontalSeparator"],
```

Comment:
Next we can have extra object type of HorizontalSeparator and not relay on the first value beeing "undefined".
Again by convention try to use "null" for empty values rather then "undefined".